salt/runners/pillar.py

```diff
@@ -97,3 +98,103 @@ def show_pillar(minion="*", **kwargs):
 
     compiled_pillar = pillar.compile_pillar()
     return compiled_pillar
+
+
+def clear_pillar_cache(minion="*", **kwargs):
+    """
+    Clears the cached values when using pillar_cache
+
+    CLI Example:
+
+    Clears the pillar cache for a specific minion:
+
+    .. code-block:: bash
+
+        salt-run pillar.clear_pillar_cache 'minion'
+
+    """
+
+    if not __opts__.get("pillar_cache"):
+        log.info("The pillar_cache is set to False or not enabled.")
+        return False
+
+    ckminions = salt.utils.minions.CkMinions(__opts__)
+    ret = ckminions.check_minions(minion)
+
+    pillarenv = kwargs.pop("pillarenv", None)
+    saltenv = kwargs.pop("saltenv", None)
+
+    pillar_cache = {}
+    for tgt in ret.get("minions", []):
+        saltenv = "base"
```

Comment:
pretty sure this saltenv needs to be cleaned up as well, since we are grabbing from kwargs above.

Reply:
Fixed.